Cubical/HITs/Hopf.agda

```diff
@@ -5,29 +5,384 @@ open import Cubical.Core.Primitives
+Border2 : (x : S¹) → (j : I) → PartialP (j ∨ (~ j)) (λ y → Border1 x j y ≃ S¹)
+Border2 x j y = Border x j y .snd
 
--- Hopf fibration using S²
-HopfS² : S² → Set
-HopfS² base = S¹
-HopfS² (surf i j) = Glue S¹ (λ { (i = i0) → _ , idEquiv S¹
-                               ; (i = i1) → _ , idEquiv S¹
-                               ; (j = i0) → _ , idEquiv S¹
-                               ; (j = i1) → _ , _ , rotIsEquiv (loop i) } )
-
--- Hopf fibration using more direct definition of the rot equivalence
-HopfS²' : S² → Set
-HopfS²' base = S¹
-HopfS²' (surf i j) = Glue S¹ (λ { (i = i0) → _ , rotLoopEquiv i0
-                                ; (i = i1) → _ , rotLoopEquiv i0
-                                ; (j = i0) → _ , rotLoopEquiv i0
-                                ; (j = i1) → _ , rotLoopEquiv i } )
-
--- Hopf fibration using suspension of S¹
 HopfSuspS¹ : SuspS¹ → Set
 HopfSuspS¹ north = S¹
 HopfSuspS¹ south = S¹
-HopfSuspS¹ (merid x i) = ua (_ , rotIsEquiv x) i
+HopfSuspS¹ (merid x j) = Glue S¹ (Border x j)
+
+-- Total space of the fibration
+TotalSpace : Set
+TotalSpace = Σ SuspS¹ HopfSuspS¹
+
+unglueb : (x : S¹) → (j : I) → Glue S¹ (Border x j) → S¹
+unglueb x j y = unglue {_} _ {Border1 x j} {Border2 x j} y
+
+-- Forward direction
+filler-1 : I → (j : I) → (y : S¹) → Glue S¹ (Border y j) → join S¹ S¹
+filler-1 i j y x = hfill (λ t → λ { (j = i0) → inl (rotInv-1 x y t)
+                                  ; (j = i1) → inr x })
+                         (inc (push (rot (unglueb y j x) (flip y)) (unglueb y j x) j)) i
+
+Hopf→Join : TotalSpace → join S¹ S¹
+Hopf→Join (north , x) = inl x
+Hopf→Join (south , x) = inr x
+Hopf→Join (merid y j , x) = filler-1 i1 j y x
+
+-- Backward direction
+Join→Hopf : join S¹ S¹ → TotalSpace
+Join→Hopf (inl x) = (north , x)
+Join→Hopf (inr x) = (south , x)
+Join→Hopf (push y x j) =
+  (merid (rot (flip y) x) j
+  , glue {_} {_} {_} {_} {Border1 (rot (flip y) x) j} {Border2 (rot (flip y) x) j}
+         (λ { (j = i0) → y ; (j = i1) → x })
+         (rotInv-2 x y j))
+
+-- Now for the homotopies, we will need to fill squares indexed by x y : S¹ with value in S¹
+-- Some will be extremeley tough, but happen to be easy when x = y = base
+-- therefore, we fill them for x = y = base and then use the connectedness of S¹ × S¹ and
+-- the discreteness of ΩS¹ to get general fillers.
+
+-- To proceed with that strategy, we first need a lemma :
+-- the sections of the trivial fibration λ (_ : S¹) (_ : S¹) → Int are constant
+
+-- this should be generalized to a constant fibration over a connected space with
+-- discrete fiber
+fibInt : S¹ → S¹ → Set
+fibInt _ _ = Int
+
+constant-loop-0 : (F : S¹ → Int) → (j : I) → (x : I) → F base ≡ F (loop x)
+constant-loop-0 F j x =
+  hfill (λ t → λ { (x = i0) → λ _ → F base
+                 ; (x = i1) → isSetInt (F base) (F base) (λ j → F (loop j)) refl t })
+        (inc (λ i → F (loop (x ∧ i)))) j
+
+constant-loop : (F : S¹ → S¹ → Int) → (x : S¹) → (y : S¹) → F base base ≡ F x y
```

Comment:
May I suggest some change to this proof? The `hcomp`s are a little unenlightening...

All you need is the property that `Int` is a set (witnessed by `isSetInt`). So any map from `S¹` into it will be const.

Reply:
I open a PR to your fork (https://github.com/loic-p/cubical/pull/1)

If I understand it right, merging it there first, and then merge here, will include that change?

Reply:
Great observation ! 
I think that ideally, we would even want to generalize that further to the following theorem (to prove in a more appropriate file) : any map from a n-connected type to a n-truncated type is constant. 
But for now, let's use your proof.